File: lib/src/common/forms/directives/abstract_control_directive.dart
library angular2.src.common.forms.directives.abstract_control_directive;

import "../model.dart" show AbstractControl;
import "package:angular2/src/facade/lang.dart" show isPresent;
import "package:angular2/src/facade/exceptions.dart" show unimplemented;

/**
 * Base class for control directives.
 *
 * Only used internally in the forms module.
 */
abstract class AbstractControlDirective {
  AbstractControl get control {
    return unimplemented();
  }

  dynamic get value {
    return isPresent(this.control) ? this.control.value : null;
  }

  bool get valid {
    return isPresent(this.control) ? this.control.valid : null;
  }

  Map<String, dynamic> get errors {
    return isPresent(this.control) ? this.control.errors : null;
  }

  bool get pristine {
    return isPresent(this.control) ? this.control.pristine : null;
  }

  bool get dirty {
    return isPresent(this.control) ? this.control.dirty : null;
  }

  bool get touched {
    return isPresent(this.control) ? this.control.touched : null;
  }

  bool get untouched {
    return isPresent(this.control) ? this.control.untouched : null;
  }

  List<String> get path {
    return null;
  }
}

File: npm/angular2/ts/src/common/forms/directives/abstract_control_directive.ts
import {AbstractControl} from '../model';
import {isPresent} from 'angular2/src/facade/lang';
import {unimplemented} from 'angular2/src/facade/exceptions';

/**
 * Base class for control directives.
 *
 * Only used internally in the forms module.
 */
export abstract class AbstractControlDirective {
  get control(): AbstractControl { return unimplemented(); }

  get value(): any { return isPresent(this.control) ? this.control.value : null; }

  get valid(): boolean { return isPresent(this.control) ? this.control.valid : null; }

  get errors(): {[key: string]: any} {
    return isPresent(this.control) ? this.control.errors : null;
  }

  get pristine(): boolean { return isPresent(this.control) ? this.control.pristine : null; }

  get dirty(): boolean { return isPresent(this.control) ? this.control.dirty : null; }

  get touched(): boolean { return isPresent(this.control) ? this.control.touched : null; }

  get untouched(): boolean { return isPresent(this.control) ? this.control.untouched : null; }

  get path(): string[] { return null; }
}
